packages/client/modules/teamDashboard/components/ProviderRow/GitLabProviderRow.tsx

```diff
@@ -70,22 +84,78 @@ const GitLabProviderRow = (props: Props) => {
   const {teamMember} = viewer
   const {integrations} = teamMember!
   const {gitlab} = integrations
-  const {auth, cloudProvider} = gitlab
-  if (!cloudProvider) return null
-  const {clientId, id: cloudProviderId, serverBaseUrl} = cloudProvider
+  const {auth, cloudProvider, sharedProviders} = gitlab
+  const connected = !!auth
+  const connectedProviderId = auth?.provider?.id
+  const availableProviders = [...(cloudProvider ? [cloudProvider] : []), ...sharedProviders]
+
+  if (availableProviders.length === 0) return null
 
   return (
     <>
-      <ProviderRow
-        connected={!!auth}
-        onConnectClick={() => openOAuth(cloudProviderId, clientId, serverBaseUrl)}
-        submitting={submitting}
-        togglePortal={togglePortal}
-        menuRef={menuRef}
-        providerName={'GitLab'}
-        providerDescription={'Use GitLab Issues from within Parabol'}
-        providerLogo={<GitLabProviderLogo />}
-      />
+      <div className='relative my-4 flex w-full shrink-0 flex-col justify-start rounded bg-white shadow-card'>
+        <div className='flex justify-start p-row-gutter pb-0'>
+          <GitLabProviderLogo />
+          <div className='flex w-full flex-col'>
+            {availableProviders.map(({id, serverBaseUrl, clientId}) => {
+              const showProvider = !connected || id === connectedProviderId
+              const isCloudProvider = cloudProvider?.id === id
+              if (!showProvider) return null
+              return (
+                <div key={id} className='flex w-full flex-row pb-4'>
+                  <RowInfo>
+                    <div className='mr-4 flex items-center align-middle font-semibold leading-6 text-slate-700'>
+                      {isCloudProvider ? 'GitLab' : serverBaseUrl.replace(/https:\/\//, '')}
+                    </div>
+                    <RowInfoCopy>
+                      {isCloudProvider
+                        ? 'Use GitLab Issues from within Parabol.'
+                        : 'Connect to your own GitLab server.'}
+                    </RowInfoCopy>
+                  </RowInfo>
+                  <ProviderActions>
+                    {!connected && (
+                      <ConnectButton
+                        onConnectClick={() => openOAuth(id, clientId, serverBaseUrl)}
+                        submitting={submitting}
+                      />
+                    )}
+                    {id === connectedProviderId && (
+                      <>
+                        {isDesktop ? (
+                          <>
+                            <div className='flex items-center pr-6'>
+                              <DoneIcon className='h-[18px] w-[18px] text-lg text-success-light' />
```

Comment:
+1 I think it's best to use the default tailwind heights where possible rather than [6px] [18px] etc

Reply:
Yes, I agree. I just did not want to modify the existing layout in this PR. It's too big already.